tests/test_discriminated_union.py

```diff
@@ -130,7 +130,18 @@ class Dog(BaseModel):
     ]
 
 
-def test_discriminated_union_validation():
+@pytest.mark.parametrize('color_discriminator_kind', ['discriminator', 'field_str', 'field_discriminator'])
+@pytest.mark.parametrize('pet_discriminator_kind', ['discriminator', 'field_str', 'field_discriminator'])
+def test_discriminated_union_validation(color_discriminator_kind, pet_discriminator_kind):
+    def _get_str_discriminator(discriminator: str, kind: str):
+        if kind == 'discriminator':
+            return Discriminator(discriminator)
+        elif kind == 'field_str':
+            return Field(discriminator=discriminator)
+        elif kind == 'field_discriminator':
+            return Field(discriminator=Discriminator(discriminator))
+        raise ValueError(f'Invalid kind: {kind}')
```

Comment:
I feel that this is a fairly thorough test that all the ways to possibly use a `str` as a discriminator, even together, are working as expected. 🤞